users/views.py

```diff
@@ -31,6 +31,26 @@ def balance(request):
 
 
 @permission_required('users.change_balance')
+def buy_product(request):
+    if request.method == 'POST':
+        form = ProductBuyForm(request.POST)
+        if form.is_valid():
```

Comment:
faudrait une transaction ici autour

Reply:
Ça pourrait valoir le coup d'utiliser https://docs.djangoproject.com/en/1.11/topics/db/transactions/#tying-transactions-to-http-requests (qui résoud le commentaire suivant aussi) ?

Reply:
Ca vaudrait la peine ouais :)
Mais ça fixe pas le commentaire suivant : ici c'est pour si on crash au milieu, l'autre c'est un pb de concurrence

Reply:
(note que j'aime bien l'idée d'une transaction explicite pour des trucs qui jouent avec de la thune)